lib/timers.js

```diff
@@ -301,6 +299,14 @@ const Timeout = function(after) {
   this._repeat = null;
 };
 
+
+function unrefdHandle() {
+  this.owner._onTimeout();
+  if (!this.owner._repeat)
+    this.owner.close();
```

Comment:
This will run `this._handle.close()` which runs `uv_close(wrap->handle__, OnClose)`. Are we sure that the list of timers attached to this `TimerWrap` instance is empty before this runs?